File: rest/src/main/java/io/sparkled/rest/FillTypeRestService.java
package io.sparkled.rest;

import io.sparkled.model.animation.fill.reference.FillTypes;

import javax.ws.rs.GET;
import javax.ws.rs.Path;
import javax.ws.rs.Produces;
import javax.ws.rs.core.MediaType;
import javax.ws.rs.core.Response;

@Path("/fill-type")
public class FillTypeRestService extends RestService {

    @GET
    @Produces(MediaType.APPLICATION_JSON)
    public Response get() {
        return getJsonResponse(FillTypes.get());
    }
}

File: rest/src/main/kotlin/io/sparkled/rest/service/filltype/FillTypeRestService.kt
package io.sparkled.rest.service.filltype

import javax.inject.Inject
import javax.ws.rs.GET
import javax.ws.rs.Path
import javax.ws.rs.Produces
import javax.ws.rs.core.MediaType
import javax.ws.rs.core.Response

@Path("/fillTypes")
class FillTypeRestService
@Inject constructor(private val handler: FillTypeRestServiceHandler) {

    @GET
    @Produces(MediaType.APPLICATION_JSON)
    fun get(): Response {
        return handler.get()
    }
}